jupyter-extension/src/plugins/mount/pollMounts.ts

```diff
@@ -129,23 +123,58 @@ export class PollMounts {
     return this._dataPoll;
   }
 
-  refresh = async (): Promise<void> => {
-    await this._dataPoll.refresh();
-    await this._dataPoll.tick;
-  };
+  updateMountedRepo = (
+    repo: Repo | null,
+    mountedBranch: Branch | null,
+  ): void => {
+    if (repo === null) {
+      localStorage.removeItem(PollMounts.MOUNTED_REPO_LOCAL_STORAGE_KEY);
+      this.mountedRepo = null;
+      return;
+    }
 
-  updateData = (data: ListMountsResponse): void => {
-    if (JSON.stringify(data) !== JSON.stringify(this._rawData)) {
-      this._rawData = data;
-      this.mounted = Array.from(Object.values(data.mounted));
-      this.unmounted = Array.from(Object.values(data.unmounted));
+    if (!mountedBranch) {
+      mountedBranch = repo?.branches[0] || null;
+      for (const branch of repo.branches) {
+        if (branch.name === 'master') {
+          mountedBranch = branch;
+        }
+      }
     }
+
+    this.mountedRepo = {
+      mountedBranch,
+      repo,
+    };
+    localStorage.setItem(
+      PollMounts.MOUNTED_REPO_LOCAL_STORAGE_KEY,
+      JSON.stringify(this.mountedRepo),
+    );
   };
 
-  updateProjects = (data: ProjectInfo[]): void => {
-    if (JSON.stringify(data) !== JSON.stringify(this.projects)) {
-      this.projects = data;
+  getMountedRepoInputSpec = (): CrossInputSpec | PfsInput => {
+    const mountedRepo = this.mountedRepo;
+    if (mountedRepo === null) {
+      return {};
+    }
+
+    let repo = mountedRepo.repo.name;
+    if (mountedRepo.repo.project !== 'default') {
+      repo = `${mountedRepo.repo.project}_name`;
     }
+
+    return {
+      pfs: {
+        name: `${mountedRepo.repo.project}_${mountedRepo.repo.name}_${mountedRepo.mountedBranch.name}`,
```

Comment:
Should this be `mountedRepo.mountedBranch?.name`?

Reply:
We check the `mountedRepo` param earlier in this method for null and return `{}` so I think this is safe as is.

Reply:
Is it possible for `mountedRepo` to be non-null and `mountedBranch` to be null?

Reply:
@smalyala nope. mountedRepo is non null when the user has selected at least a project/repo. Selecting that will automatically select a branch (usually master). It can only be changed to other projects, repos, branches with no option to manually unselect just a branch.